devito/symbolics/extended_sympy.py

```diff
@@ -544,6 +548,9 @@ def __str__(self):
 
     __repr__ = __str__
 
+    def _sympystr(self, printer):
```

Comment:
What's that?

Reply:
eheh

SymPy dark voodoo magic :)

try a `git grep _sympystr` in sympy, you will see

basically a magic method that one needs to override to make sure Basic subclasses emit their actual `repr` when part of a larger expression

E.g.

Say you have an `IntDiv(a, 4)`. When you are in the terminal, you want to see `a / 4` (or `a // 4`, whatever), not `IntDiv(a, 4)`. This works in master... but only if not part of a larger expression. For example, we would see `IntDiv(a, 4) + 3`. Overriding `_sympystr` allows you to display `(a / 4) + 3`. See the tests that I've added